drivers/serial/uart_nrfx_uarte.c

```diff
@@ -607,17 +672,20 @@ static int uarte_instance_init(struct device *dev,
 	#endif /* CONFIG_UART_0_INTERRUPT_DRIVEN */
 
 	#ifdef CONFIG_UART_0_NRF_FLOW_CONTROL
-		#define UARTE_0_NRF_HWFC_CONFIG	       UARTE_NRF_HWFC_ENABLED(0)
-		#ifndef DT_NORDIC_NRF_UARTE_UART_0_RTS_PIN
-		#error Flow control for UARTE0 is enabled, but RTS pin is not defined.
-		#endif
-		#ifndef DT_NORDIC_NRF_UARTE_UART_0_CTS_PIN
-		#error Flow control for UARTE0 is enabled, but CTS pin is not defined.
-		#endif
+		#define UARTE_0_NRF_HWFC_CONFIG NRF_UARTE_HWFC_ENABLED
 	#else
-		#define UARTE_0_NRF_HWFC_CONFIG	       UARTE_NRF_HWFC_DISABLED
+		#define UARTE_0_NRF_HWFC_CONFIG NRF_UARTE_HWFC_DISABLED
 	#endif /* CONFIG_UART_0_NRF_FLOW_CONTROL */
 
+	#if defined(DT_NORDIC_NRF_UARTE_UART_0_RTS_PIN) && \
+	    defined(DT_NORDIC_NRF_UARTE_UART_0_CTS_PIN)
+		#define UARTE_0_NRF_RTS_CTS_PINS UARTE_NRF_RTS_CTS_SET(0)
+		#define UARTE_0_CONFIG_RTS_CTS .rts_cts_pins_set = true
```

Comment:
You could add more spaces after macros names and start definitions in the same column. To keep consistency with the rest and improve readability. Same story right below and in corresponding macros for UARTE_1.